src/Rings/groebner.jl

```diff
@@ -61,39 +60,69 @@ returned in `B.S`.
 julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"])
 (Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
 
-julia> A = Oscar.BiPolyArray([x*y-3*x,y^3-2*x^2*y])
-Oscar.BiPolyArray{fmpq_mpoly}(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x*y - 3*x, -2*x^2*y + y^3], #undef, #undef, false, #undef, true)
+julia> A = Oscar.IdealGens([x*y-3*x,y^3-2*x^2*y])
+Oscar.IdealGens{fmpq_mpoly}(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x*y - 3*x, -2*x^2*y + y^3], #undef, #undef, false, #undef, true)
 
-julia> B = Oscar._compute_groebner_basis(A, degrevlex(gens(R)))
-Oscar.BiPolyArray{fmpq_mpoly}(Multivariate Polynomial Ring in x, y over Rational Field, #undef, Singular Polynomial Ring (QQ),(x,y),(dp(2),C), Singular ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (x*y - 3*x, y^3 - 6*x^2, 2*x^3 - 9*x), true, #undef, true)
+julia> B = Oscar._compute_std_basis(A, degrevlex(gens(R)))
+Oscar.IdealGens{fmpq_mpoly}(Multivariate Polynomial Ring in x, y over Rational Field, #undef, Singular Polynomial Ring (QQ),(x,y),(dp(2),C), Singular ideal over Singular Polynomial Ring (QQ),(x,y),(dp(2),C) with generators (x*y - 3*x, y^3 - 6*x^2, 2*x^3 - 9*x), true, #undef, true)
 ```
 """
-function _compute_groebner_basis(B::BiPolyArray, ordering::MonomialOrdering, complete_reduction::Bool = false, enforce_global_ordering::Bool = true)
+function _compute_std_basis(B::IdealGens, ordering::MonomialOrdering, complete_reduction::Bool = false)
    singular_assure(B, ordering)
    R = B.Sx
-   if enforce_global_ordering
-     !Oscar.Singular.has_global_ordering(R) && error("The ordering has to be a global ordering.")
-   end
    I  = Singular.Ideal(R, gens(B.S)...)
    i  = Singular.std(I, complete_reduction = complete_reduction)
-   BA = BiPolyArray(B.Ox, i)
+   BA = IdealGens(B.Ox, i)
    BA.isGB  = true
+   BA.ord = ordering
    if isdefined(BA, :S)
        BA.S.isGB  = true
    end
 
    return BA
 end
 
+# standard basis for non-global orderings #############################
+@doc Markdown.doc"""
+    std_basis(I::MPolyIdeal, ordering::MonomialOrdering)
+
+Compute a standard basis of `I` for the monomial ordering `ordering`.
+
+**Note:** Since there is no general notion of complete reduction for 
+non-global orderings, this option is not available for this command;
+instead, use `groebner_basis` directly. 
+
+# Examples
+```jldoctest
+julia> R,(x,y) = PolynomialRing(QQ, ["x","y"])
+(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
+
+julia> I = ideal([x*(x+1), x^2-y^2+(x-2)*y])
+ideal(x^2 + x, x^2 + x*y - y^2 - 2*y)
+
+julia> std_basis(I, negdegrevlex(gens(R)))
+2-element Vector{fmpq_mpoly}:
+ x
+ y
+```
+"""
+function std_basis(I::MPolyIdeal, ordering::MonomialOrdering, complete_reduction::Bool = false)
```

Comment:
In the other Gröbner related functions we have, in general, `ordering` as a keyword argument.